### PR TITLE
Add a null/no-op volume driver

### LINKED ISSUE #153: develop a null volume driver
This driver shall be primarily used for testing and benchmarking voplugin functionality in isolation.

### PR DESCRIPTION
**Pull request by mapuri**
The changes here introduce a null driver that basically doesn't perform any actual disk operations.

The driver to use for a volume is set in `volume-options` using field called `backend`. The default backend is `ceph` and a user can override it by specifying the `backend` field in `default-options` structure.

The primary use case of the null driver will be to use it in scale/battery tests to exercise/benchmark volplugin alone  and I plan to work on that next. But I put out this PR to make sure I am working in right direction and also to keep the patch small.

fixes #153 

@erikh @unclejack PTAL

### CODE REVIEW

File: systemtests/util_test.go [hunk 1]
```diff
@@ -31,7 +32,11 @@ func (s *systemtestSuite) readIntent(fn string) (*config.TenantConfig, error) {
 		return nil, err
 	}
 
-	cfg := &config.TenantConfig{}
+	cfg := &config.TenantConfig{
```
Comment (erikh) [inline systemtests/util_test.go]: since these are only used for tests, let's change the testdata instead of adding this code.

File: volcli/volcli.go [hunk 2]
```diff
@@ -133,7 +134,11 @@ func tenantUpload(ctx *cli.Context) (bool, error) {
 		return false, err
 	}
 
-	tenant := &config.TenantConfig{}
+	tenant := &config.TenantConfig{
```
Comment (erikh) [inline volcli/volcli.go]: there's a lot of repetition about this. Let's instead make  amethod, `config.NewTenantConfig(driver string)` that performs this construction. This can also be used in the tests where you do this.

Does this work for you?

**Comment by erikh** [comment]
Final comment: I would like to see the notion of a driver owning its own name -- can we add a `Name() string` method to the interface and each of the drivers that returns the right thing? It could be useful for diagnostic purposes later.

**Comment by erikh** [comment]
Oh. The notion of a global default would be very nice, if you have the time to incorporate that.

File: systemtests/util_test.go (hunk 1 above)
Comment (mapuri) [inline systemtests/util_test.go]: yes, we can do this.

Or see my next comment, may using `config.NewDefaultTenantConfig()` here as well might be better as that will help us test that `ceph` is picked as default driver when one is not specified by the user.

File: volcli/volcli.go (hunk 2 above)
Comment (mapuri) [inline volcli/volcli.go]: yeah, I agree the repetition can be avoided by using a function for default config.

I am wondering if having `config.NewDefaultTenantConfig()` method will be better which will return an empty config with `backend` set to `ceph`. Does it sound right?

Later we can may be add other things to this method as well like I saw in volcli tests we by-default add a `"ext4": "mkfs.ext4 -m0 %"` filesystem to the config options. But I might be wrong about it's purpose.

**Comment by mapuri** [comment]
thanks for the review @erikh 

> can we add a Name() string method to the interface and each of the drivers that returns the right thing?

that is a good idea, let me try it out.

> The notion of a global default would be very nice, if you have the time to incorporate that.

I am not sure if I completely understood this. By global-default do you imply having ceph as a default backed. If so I am wondering if `config.NewDefaultTenantConfig()` function as suggested above will address this?

Let me work on a patch with this change in a bit as it might be more clear that way.

**Comment by erikh** [comment]
@mapuri I mean a global default as in the new "global configuration" (see examples in `systemtests/testdata/global1.json`, which would revert to ceph if unspecified.

**Comment by erikh** [comment]
and yes, a perfect place to consume this would be in your new function.

**Comment by mapuri** [comment]
@erikh this should be ready for another look. Following is the highlight of changes:

- Added `config.NewTenantConfig` function to return a tenant config with specified backend name
- Added `config.NewGlobalConfig` function to return a global config with default backend name and ttl
- In tenant upload volcli use the backend name from global-config as a default when it is not provided by the user.
 - also fixed global config related volcli handlers to be consistent with other cli handlers
- added `Name()` method to `storage.Driver` interface which can be used to return the backend name of a gvien instance.

**Comment by erikh** [comment]
LGTM. Thanks for the extra work on globals.

**Comment by erikh** [comment]
let's wait for @unclejack to peek at it before merging.

**Comment by erikh** [comment]
I'm going to merge #163 before this, are you comfortable rebasing against this @mapuri ?

File: .gitignore [hunk 3]
```diff
@@ -13,3 +13,5 @@ node_modules
 dist
 docs/docs
 .DS_Store
+*.swp
+cscope.*
```
Comment (unclejack) [inline .gitignore]: @mapuri This change is unrelated to the commits.

**Comment by unclejack** [comment]
LGTM with one nit to address: the extraneous change made to `.gitignore`.

File: .gitignore (hunk 3 above)
Comment (mapuri) [inline .gitignore]: @unclejack yes, you are right. It is needed to avoid showing my ide and tags files in git status. I just slipped it in with this PR to avoid having to create a separate issue. Let me remove it and create a separate PR.

**Comment by mapuri** [comment]
@erikh yes, should be fine. I will rebase once you have merged the other PR.

File: .gitignore (hunk 3 above)
Comment (erikh) [inline .gitignore]: I dunno if this is necessary, let's just move forward with this patch.

Rebase is already going to be hell. :)

**Comment by erikh** [comment]
@mapuri rebase please :)

**Comment by mapuri** [comment]
yep, will take care of this in a bit :)

**Comment by mapuri** [comment]
@erikh I have rebased the patch. PTAL when you get a chance.

**Comment by mapuri** [comment]
uh..oh, don't bother, looks like it was already merged :)
